msticpy/context/azure/sentinel_ti.py

```diff
@@ -0,0 +1,455 @@
+        external_references : list, optional
+            A list of URLs that provide an external reference for the indicator
+        valid_from : datetime, optional
+            A datetime from which the indicator is valid from, defaults to now
+        valid_to : datetime, optional
+            A datetime to which the indicator is valid until
+
+        Raises
+        ------
+        CloudError
+            If API call fails
+
+        """
+        self.check_connected()  # type: ignore
+        ti_url = self.sent_urls["ti"] + f"/indicators/{indicator_id}"  # type: ignore
+        indicator_details = self.get_indicator(indicator_id)
+        data_items = _build_additional_indicator_items(**kwargs)
+        data_items.pop("validFrom")
+        full_data_items = _add_missing_items(data_items, indicator_details)
+        data = _build_sent_data(full_data_items, props=True)
+        data["etag"] = indicator_details["etag"]
+        data["kind"] = "indicator"
+        params = {"api-version": "2021-10-01"}
+        response = httpx.put(
+            ti_url,
+            headers=get_api_headers(self.token),  # type: ignore
+            params=params,
+            content=str(data),
+            timeout=get_http_timeout(),
+        )
+        if response.status_code not in (200, 201):
+            raise CloudError(response=response)
+        print("Indicator updated.")
+
+    def add_tag(self, indicator_id: str, tag: str):
+        """
+        Add a tag to an existing indicator.
+
+        Parameters
+        ----------
+        indicator_id : str
+            The GUID of the indicator to add a tag to.
+        tag : str
+            The tag to add.
+
+        """
+        self.check_connected()  # type: ignore
+        indicator_details = self.get_indicator(indicator_id)
+        tags = [tag]
+        if "threatIntelligenceTags" in indicator_details["properties"]:
+            tags += indicator_details["properties"]["threatIntelligenceTags"]
+        self.update_indicator(indicator_id=indicator_id, labels=tags)
+
+    def delete_indicator(self, indicator_id: str):
+        """
+        Delete a specific TI indicator.
+
+        Parameters
+        ----------
+        indicator_id : str
+            The GUID of the indicator to delete
+
+        Raises
+        ------
+        CloudError
+            If API call fails
+
+        """
+        self.check_connected()  # type: ignore
+        ti_url = self.sent_urls["ti"] + f"/indicators/{indicator_id}"  # type: ignore
+        params = {"api-version": "2021-10-01"}
+        response = httpx.delete(
+            ti_url,
+            headers=get_api_headers(self.token),  # type: ignore
+            params=params,
+            timeout=get_http_timeout(),
+        )
+        if response.status_code not in (200, 204):
+            raise CloudError(response=response)
+        print("Indicator deleted.")
+
+    def query_indicators(self, **kwargs) -> pd.DataFrame:
```

Comment:
Seems odd that you cannot query for an indicator value (i.e. like aa IP address)

Reply:
So you can use the 'keyword' part of the query here if you want to do that. I will make it clearer in the docs. Honestly I don't think this feature is really that useful. I added it as there is an API for it but given the constrained query langauge I don't know why you would use is over runiing KQL against the TIIndicators table.